### ISSUE TITLE
Ability to install mocks for interfaces with QuarkusMock

### ISSUE THREAD

**Issue by josephearl** [quarkusio/quarkus#29706]
### Description

I have a plain Java interface e.g. `StoreInventory`, a real implementation `DbStoreInventory` using Panache that is `@ApplicationScoped` and `ProductResource` that depends on `StoreInventory`.

I am trying do something like the following with a fake `StoreInventory` implementation `InMemoryStoreInventory`:

```
@QuarkusTest
@TestHttpEndpoint(ProductResource.class)
class ProductResourceTest {
  @Inject StoreInventory inventory;

  @BeforeAll
  static void setup() {
    QuarkusMock.installMockForType(new InMemoryStoreInventory(), StoreInventory.class);
  }
}
```

then I get an (erroneous) error:

```
DbStoreInventory@13cc31df is not a normal scoped CDI bean, make sure the bean is a normal scope like @ApplicationScoped or @RequestScoped
```

checking the cause shows the real issue:

```
Caused by: java.lang.ClassCastException: class InMemoryStoreInventory cannot be cast to class DbStoreInventory
```

**Comment by geoand** [comment]
> I  am able to use @InjectMock to inject a new Mockito mock for StoreInventory, but perhaps this is actually just mocking DbStoreInventory?

Correct.

In your case, you can likely use `@Mock` as described [here](https://quarkus.io/guides/getting-started-testing#cdi-alternative-mechanism).

**Comment by josephearl** [comment]
Thanks, but my understanding is `@Mock` would apply to all tests, which I don’t want

**Comment by geoand** [comment]
Indeed. So this request sounds a lot like #29707. I think we can close this one as a duplicate, WDYT?

**Comment by josephearl** [comment]
Yes, both are related. This was asking for a programmatic way of doing it, #29707 for a declarative way. Happy for you to close this one.

**Comment by geoand** [comment]
Thanks

**Comment by kpagratis** [comment]
We are also running into this problem. Could we get an update since #29707 was closed without resolution?

**Comment by geoand** [comment]
@kpagratis what's your use case exactly?
